File: RecycleSearch/app/src/main/java/br/com/vinicius/recyclesearch/ExampleAdapter.java
package br.com.vinicius.recyclesearch;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

class ExampleAdapter extends RecyclerView.Adapter<ExampleAdapter.Holder> {

    List<String> lista;

    public ExampleAdapter(List<String> lista) {
        this.lista = lista;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View exampleAdapter = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.adapter_example, viewGroup, false);
        return new Holder(exampleAdapter);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int i) {
        holder.textView.setText(lista.get(i));
    }

    @Override
    public int getItemCount() {
        if(lista == null) return 0;
        return lista.size();
    }

    public void filterList(ArrayList<String> filteredList) {
        lista = filteredList;
        notifyDataSetChanged();
    }

    public class Holder extends RecyclerView.ViewHolder{

        private TextView textView;

        public Holder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.textView);
        }

    }
}

File: RecycleSearch/app/src/main/java/br/com/vinicius/recyclesearch/ExampleAdapter.kt
package br.com.vinicius.recyclesearch

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList

internal class ExampleAdapter(var lista: List<String>?) : RecyclerView.Adapter<ExampleAdapter.Holder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): Holder {
        val exampleAdapter = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.adapter_example, viewGroup, false)
        return Holder(exampleAdapter)
    }

    override fun onBindViewHolder(holder: Holder, i: Int) {
        holder.textView.text = lista!![i]
    }

    override fun getItemCount(): Int {
        return if (lista == null) 0 else lista!!.size
    }

    fun filterList(filteredList: ArrayList<String>) {
        lista = filteredList
        notifyDataSetChanged()
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val textView: TextView

        init {
            textView = itemView.findViewById(R.id.textView)
        }

    }
}
